src/qt/clientmodel.cpp

```diff
@@ -226,31 +226,39 @@ static void BannedListChanged(ClientModel *clientmodel)
     assert(invoked);
 }
 
-static void BlockTipChanged(ClientModel *clientmodel, bool initialSync, int height, int64_t blockTime, double verificationProgress, bool fHeader)
+static ClientModel::NotificationStatus GetNotificationStatus (interfaces::Node& node, bool initial_sync)
 {
-    // lock free async UI updates in case we have a new block tip
-    // during initial sync, only update the UI if the last update
-    // was > 250ms (MODEL_UPDATE_DELAY) ago
-    int64_t now = 0;
-    if (initialSync)
-        now = GetTimeMillis();
-
-    int64_t& nLastUpdateNotification = fHeader ? nLastHeaderTipUpdateNotification : nLastBlockTipUpdateNotification;
+    if (node.getImporting()) return ClientModel::NotificationStatus::INIT_IMPORT;
```

Comment:
I'm not sure what motivation is exposing the import state, but it seems like you'd want the state to be `INIT_REINDEX` not `INIT_IMPORT` during reindexing (see [`ThreadImport`](https://github.com/bitcoin/bitcoin/blob/e5d47ed8fd3724cc5295fca70573f1edf552ff62/src/init.cpp#L676-L693)) so I don't understand the way these are prioritized

Reply:
Reworked.